util/plot_utils.py

```diff
@@ -4,20 +4,52 @@
 """
 import torch
 import pandas as pd
-from pathlib import Path
 import seaborn as sns
 import matplotlib.pyplot as plt
 
+from pathlib import Path, PurePath
+from typing import Iterable
+import copy
 
-def plot_logs(logs, fields=('class_error', 'loss_bbox_unscaled', 'mAP'), ewm_col=0):
-    dfs = [pd.read_json(Path(p) / 'log.txt', lines=True) for p in logs]
+
+def plot_logs(logs, fields=('class_error', 'loss_bbox_unscaled', 'mAP'), ewm_col=0, log_name='log.txt'):
+    '''
+    Function to plot specific fields from training logs
+    :: Inputs - logs = list containing one or more directories, each w/ single log file (convert to list if needed)
+              - fields = which training results to plot from log file
+              - ewm_col = ??
+              - log_name = optional, name for log file(s) if different than default 'log.txt'
+    :: Outputs - matplotlib plots of items in fields per each log file.
+               - solid lines are training results, dashed lines are test results
+    '''
+
+    # verify logs is list, convert if not. handle single dir, etc. *Should not have side-effects, makes deep new logs
+    if not isinstance(logs, list):
```

Comment:
I think if we want to be more robust to the user, we might want to either enforce that it should be a `Path` or a `List[Path]`, but otherwise it can become more complicated

Reply:
I agree re: becoming more complicated!  (...I had started thinking about all the variety of incoming items that could be passed in, and it goes downhill fast).

Let me make a couple changes based on your feedback above, test those, and will update tonight. 
Thanks much for the review and feedback!

Reply:
I've simplified it to check for either a Path, or List[Path] now per your recommendation.

First check if list, verify if Path type, then convert to list for single Path item.
Then iterate list and check each item is a)Path type and b)verify dir exists...

and then onto plotting.

I tested a bunch of combos to verify the code, and will submit tomorrow (edit - found issue with tuples...fixing).  

Question - is there a specific VScode lint plugin you use for verifying/formatting to pass your CircleCI?  
I was using one this weekend (flake8), but I still had to fix about 4 or 5 whitespace issues by hand, so wanted to see if there is a specific format plugin I should run?

Thanks much!

Reply:
> is there a specific VScode lint plugin you use for verifying/formatting to pass your CircleCI?
> I was using one this weekend (flake8), but I still had to fix about 4 or 5 whitespace issues by hand, so wanted to see if there is a specific format plugin I should run?

We just use [flake8 with the some extra custom configs](https://github.com/facebookresearch/detr/blob/master/tox.ini), so I believe you just need to point VSCode to use that and it should be enough

Reply:
Thanks @fmassa  - I've updated the doc string section, and simplified the argument pre-checks to exclusively support the simpler Path and list[Path] scenarios.
I did test a number of random inputs (strings, tuples, mixes of Path and string) etc. to try and be robust on the testing as well as verify error messages are informative.

So now it supports single dir scenario, and provides err info for anything outside the bounds of supported args, as well as directory verification with info for any incorrect ones.

I also like having the log_name as an exposed, optional parameter rather than being internally hardcoded.

btw - The plots generated from this function are quite nice!